Kernel/Inference.cpp

```diff
@@ -398,7 +398,7 @@ Inference::Inference(const FormulaTransformation& ft) {
   init1(ft.rule,ft.premise);
 
   ASS_REP(isFormulaTransformation(ft.rule),ruleName(ft.rule));
-  ASS(!ft.premise->isClause());
+  // ASS(!ft.premise->isClause()); // TODO: should this assertion be here? it would keeps us from preprocessing clauses in InterpretedNormalizer
```

Comment:
The idea was: if you apply this to clauses, don't call it a FormulaTransformation.

Do you have a use case for calling FormulaTransformation on clauses? (I mean, it sounds sensible, but if that's the case, let's maybe think if we can single it out and assign it a different inference rule as well!)

Reply:
Ahhh i see. The reason is that I used InterpretdNormalization for Clauses as well as for formulas at some point (twice in the preprocessing chain). This is not the case anymore so adding the assertion back is fine i think. I just tried this and doing some debug runs now to make sure everything s still fine.

Reply:
So the ALASCA preprocessing is done after clausification. I didn't find a `ClauseTransformation` so I thought `FormulaTransformation` is the inference to go with. Or should I use `SimplifyingInference1` you reckon? Feels not quite right.

Reply:
I think we could add `ClauseTransformation`. I can do it if you agree that that's what feels like the right thing!